src/VisualStudio/CSharp/Test/Interactive/Commands/ResetInteractiveTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Test.Utilities;
+using Xunit;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Editor.Host;
+using System.Collections.Generic;
+using System;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Interactive.Commands
+{
+    public class ResetInteractiveTests
+    {
+        private string WorkspaceXmlStr =>
+@"<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"">
+        <Document FilePath=""CSharpDocument""></Document>
+    </Project>
+</Workspace>";
+
+        [WpfFact]
+        [Trait(Traits.Feature, Traits.Features.Interactive)]
+        public async void TestResetREPLWithProjectContext()
+        {
+            using (var workspace = await TestWorkspace.CreateAsync(WorkspaceXmlStr))
+            {
+                var expectedSubmissions = new List<string> {
+                    "r:r1\r\nr:r2\r\n",
+                    "i:ns1\r\ni:ns2\r\n"};
+                AssertResetInteractive(workspace, buildSucceeds: true, expectedSubmissions: expectedSubmissions);
```

Comment:
Consider testing a case with a project-to-project reference.  Also, maybe something portable.